### PR TITLE
WebElement now knows its origin

### PR DESCRIPTION
**Pull request by mfalesni**
Until now the `WebElement` objects were in logs and they did not show any useful informations. Now they do. Any WebElement created through `elements()` knows where did it come from and its `__repr__` accounts for that appropriately:

```python
In [5]: sel.element(".//a", root=sel.element("//li"))
Out[5]: element(('xpath', './/a'), root=element(('xpath', '//li')))
```

+ *(not sure about this feature because it might resolve to another element if there are more)* the `elements` function now does not copy the element if it received `WebElement` but it actually creates a new one based on the source locators (if present). The repr might be not as useful if you use callables in them, because they obscure :)

*I'd love to see something like this in the upstream selenium :)*

### CODE REVIEW

File: cfme/fixtures/pytest_selenium.py [hunk 1]
```diff
@@ -49,6 +49,24 @@
...
+        this, parent = self._source_locator
+        if parent:
+            return "element({}, root={})".format(repr(this), repr(parent))
+        else:
+            return "element({})".format(repr(this))
+    else:
+        return _old__repr__(self)
+
+
+WebElement.__repr__ = __repr__
```
Comment (seandst) [inline cfme/fixtures/pytest_selenium.py]: Maybe guard this against reloads, something like `if WebElement.__repr__ is not __repr__`?

**Comment by dajoRH** [comment]
Lint report for commit f4642529a0930e3f2bb5973c7dafd8404afa7946:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

File: cfme/fixtures/pytest_selenium.py (hunk 1 above)
Comment (mfalesni) [inline cfme/fixtures/pytest_selenium.py]: good point. Will change that after PR tester run.

File: cfme/fixtures/pytest_selenium.py [hunk 2]
```diff
@@ -49,6 +49,24 @@
 class_selector = re.compile(r"^(?:[a-zA-Z][a-zA-Z0-9]*)?(?:[#.][a-zA-Z0-9_-]+)+$")
 
 
+# Monkeypatching WebElement
+_old__repr__ = WebElement.__repr__
+
+
+def __repr__(self):
+    if hasattr(self, "_source_locator"):
+        this, parent = self._source_locator
```
Comment (seandst) [inline cfme/fixtures/pytest_selenium.py]: I'm thinking a  `try/except AttributeError` might be a tiny bit cleaner (`_source_locator` is only looked up once), but feel free to ignore this comment if you want. :)

**Comment by dajoRH** [comment]
Lint report for commit b00afffbbbd592bf0b2a258456341a5d0f780e40:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Pull request merged**